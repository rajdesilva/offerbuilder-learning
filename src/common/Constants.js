import { createIntl, createIntlCache } from "@formatjs/intl";
import * as languages from "../langs/";
// This is optional but highly recommended
// since it prevents memory leak
const cache = createIntlCache();

// Create the `intl` object
const intl = createIntl(
  {
    // Locale of the application
    // TODO:
    locale: "en",
    // Locale of the fallback defaultMessage
    defaultLocale: "en",
    messages: languages["en"],
  },
  cache
);

export const appConstants = {
  DEBOUNCE_TIME: 2000,
  MAXIMUM_CART_ITEMS_COUNT: 25,
  OFFER_STATUS: {
    ARCHIVED: "ARCHIVED",
    PUBLISHED: "PUBLISHED",
    UNPUBLISHED: "UNPUBLISHED",
  },
  OFFER_STATUS_ACTION: {
    ARCHIVE: "ARCHIVE",
    PUBLISH: "PUBLISH",
    UNPUBLISH: "UNPUBLISH",
  },
  OFFER_STATUS_OPTIONS: [
    {
      id: "UNPUBLISHED",
      name: "UNPUBLISHED",
    },
    {
      id: "PUBLISHED",
      name: "PUBLISHED",
    },
    {
      id: "ARCHIVED",
      name: "ARCHIVED",
    },
  ],
  offerListTab: {
    ACTIVE: "active",
    ARCHIVE: "archive",
  },
  USER_ROLE: {
    ADMIN: "offerbuilder.admin",
    EDITOR: "offerbuilder.editor",
    VIEWER: "offerbuilder.viewer",
  },
  PROPERTY_TYPE_LIST: [
    {
      name: 'DEMO',
      id: 'DEMO'
    },
    {
      name: 'PROD',
      id: 'PROD'
    },
    {
      name: 'ALL (DEMO + PROD)',
      id: 'ALL'
    },
  ],
  DISPLAY_USER_ROLE: {
    "offerbuilder.admin": intl.formatMessage({
      id: "nemo.admin",
    }),
    "offerbuilder.editor": intl.formatMessage({
      id: "nemo.editor",
    }),
    "offerbuilder.viewer": intl.formatMessage({
      id: "nemo.viewer",
    }),
  },
  USER_STATUS: {
    ACTIVE: "ACTIVE",
    INACTIVE: intl.formatMessage({
      id: "nemo.userInactiveStatus",
    }),
    NOTREGISTERED: intl.formatMessage({
      id: "nemo.notRegisteredUser",
    }),
  },
  OFFER_IMAGE: {
    ALL: "ALL",
    EXCLUDE: "EXCLUDE",
    INCLUDE: "INCLUDE",
  },
  SORT_PROPERTY: {
    NAME_ASCENDING: intl.formatMessage({
      id: "nemo.nameAscending",
    }),
    PRICE_ASCENDING: intl.formatMessage({
      id: "nemo.priceAscending",
    }),
    NAME_DESCENDING: intl.formatMessage({
      id: "nemo.nameDescending",
    }),
    PRICE_DESCENDING: intl.formatMessage({
      id: "nemo.priceDescending",
    }),
    NAME_ASCENDING_VALUE: {
      sortOrder: "ASCENDING",
      sortCriteria: "NAME",
    },
    NAME_DESCENDING_VALUE: {
      sortOrder: "DESCENDING",
      sortCriteria: "NAME",
    },
    PRICE_ASCENDING_VALUE: {
      sortOrder: "ASCENDING",
      sortCriteria: "PRICE",
    },
    PRICE_DESCENDING_VALUE: {
      sortOrder: "DESCENDING",
      sortCriteria: "PRICE",
    },
  },
};
