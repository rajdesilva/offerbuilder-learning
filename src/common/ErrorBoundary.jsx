import React from "react";
import { FormattedMessage } from "react-intl";
import styles from './css/ErrorBoundary.module.less';
class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  componentDidCatch(error, info) {
    this.setState({ hasError: true });
    window.apm.captureError(error);
    window.apm.captureError(info);
  }

  render() {
    if (this.state.hasError) {
      return (
        <div className={styles.errorBoundary}>
          <h2>
            <FormattedMessage id="nemo.somethingWentWrongPleaseTryAgain" />
          </h2>
        </div>
      );
    }
    return this.props.children;
  }
}

export default ErrorBoundary;
