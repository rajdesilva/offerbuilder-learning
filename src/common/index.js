export { default as Loader } from "./Loader";
export { appConstants } from "./Constants";
export { AccessDenied } from "./AccessDenied";
export { UserConfirmation } from "./UserConfirmation";
