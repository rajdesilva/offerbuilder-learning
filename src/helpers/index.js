export * from './api-config';
export * from './history';
