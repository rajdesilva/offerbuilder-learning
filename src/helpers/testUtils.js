import React from "react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { render as rtlRender } from "@testing-library/react";
import { getValue } from "./utility";
import rootReducers from "../redux/rootReducers";
import * as languages from "../langs";
import { IntlProvider } from "react-intl";
const initialReducerState = {};
window.getValue = getValue;

window.google = {
  maps: {
    Marker: class {},
    Map: class {},
    LatLngBounds: class {},
    places: {
      Autocomplete: class {},
      AutocompleteService: class {},
      PlacesServiceStatus: {
        INVALID_REQUEST: "INVALID_REQUEST",
        NOT_FOUND: "NOT_FOUND",
        OK: "OK",
        OVER_QUERY_LIMIT: "OVER_QUERY_LIMIT",
        REQUEST_DENIED: "REQUEST_DENIED",
        UNKNOWN_ERROR: "UNKNOWN_ERROR",
        ZERO_RESULTS: "ZERO_RESULTS",
      },
      PlacesAutocomplete: {
        INVALID_REQUEST: "INVALID_REQUEST",
        NOT_FOUND: "NOT_FOUND",
        OK: "OK",
        OVER_QUERY_LIMIT: "OVER_QUERY_LIMIT",
        REQUEST_DENIED: "REQUEST_DENIED",
        UNKNOWN_ERROR: "UNKNOWN_ERROR",
        ZERO_RESULTS: "ZERO_RESULTS",
      },
    },

    MarkerClusterer: class {},
    Geocoder: class {},
  },
};
Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

function render(
  ui,
  {
    initialState = initialReducerState,
    store = createStore(rootReducers, initialState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <IntlProvider locale="en" messages={languages["en"]}>
          {children}
        </IntlProvider>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from "@testing-library/react";

// override render method
export { render };
