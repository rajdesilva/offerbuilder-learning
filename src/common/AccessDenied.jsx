import { Button } from "antd";
import React from "react";
import { FormattedMessage } from "react-intl";
import { useLocation } from "react-router-dom";
import { baseURL } from "../helpers";
import styles from './css/AccessDenied.module.less';

export function AccessDenied() {
  const query = new URLSearchParams(useLocation().search);
  const emailId = query.get("email");
  return (
    <div className={styles.accessDenied}>
      <div className={styles.screenTitle}>
        <FormattedMessage id="nemo.accessDenied" />
      </div>
      <div className={styles.deniedMsg}>
        <FormattedMessage
          id="nemo.notHavAccessToApplication"
          values={{
            email: emailId,
          }}
        />
        <br />
        <br />
        <FormattedMessage id="nemo.contactSystmeAdmin" />
      </div>
      <Button
        type="primary"
        data-testid="denied-logout-btn"
        className={styles.deniedLogoutBtn}
        onClick={() => (window.location.href = `${baseURL.API_URL_BASE}logout`)}
      >
        <FormattedMessage id="nemo.logout" />
      </Button>
    </div>
  );
}
