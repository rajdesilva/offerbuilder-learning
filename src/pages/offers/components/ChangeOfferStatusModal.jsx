import React, { Fragment, useState } from "react";
import Modal from "antd/lib/modal/Modal";
import { InfoCircleOutlined } from "@ant-design/icons";
import { FormattedMessage, useIntl } from "react-intl";
import { Button, message } from "antd";
import { updateOffer } from "../service";
import { appConstants } from "../../../common";
import styles from "./css/ChangeOfferStatusModal.module.less";

export default function ChangeOfferStatusModal({
  offerInfo,
  actionType,
  hideModal,
}) {
  const intl = useIntl();
  const [visible, setVisible] = useState(true);
  const [loading, setLoading] = useState(false);
  const handleOk = () => {
    setLoading(true);
    if (actionType === appConstants.OFFER_STATUS_ACTION.ARCHIVE) {
      updateOffer(offerInfo.offerId, appConstants.OFFER_STATUS.ARCHIVED)
        .then((response) => {
          setLoading(false);
          if (window.getValue(response, "success")) {
            message.success(
              intl.formatMessage({
                // Matching ID as above
                id: "nemo.archivedOfferSuccessfully!",
                // Default Message in English
              })
            );
            hideModal(true);
            setVisible(false);
          } else {
            if (window.getValue(response, "error.message")) {
              message.error(window.getValue(response, "error.message"));
            } else {
              message.error(
                intl.formatMessage({
                  // Matching ID as above
                  id: "nemo.failedArchiveMsg!",
                  // Default Message in English
                })
              );
            }
          }
        })
        .catch((error) => {
          console.log("error = ", error.toString());
          message.error(
            intl.formatMessage({
              // Matching ID as above
              id: "nemo.unknownError",
              // Default Message in English
            })
          );
        });
    } else if (actionType === appConstants.OFFER_STATUS_ACTION.PUBLISH) {
      updateOffer(offerInfo.offerId, appConstants.OFFER_STATUS.PUBLISHED)
        .then((response) => {
          setLoading(false);
          if (window.getValue(response, "success")) {
            message.success(
              intl.formatMessage({
                // Matching ID as above
                id: "nemo.publishedOfferSuccessfully!",
                // Default Message in English
              })
            );
            hideModal(true);
            setVisible(false);
          } else {
            if (window.getValue(response, "error.message")) {
              message.error(window.getValue(response, "error.message"));
            } else {
              message.error(
                intl.formatMessage({
                  // Matching ID as above
                  id: "nemo.failedToPublishOffer",
                  // Default Message in English
                })
              );
            }
          }
        })
        .catch((error) => {
          console.log("error = ", error.toString());
          message.error(
            intl.formatMessage({
              // Matching ID as above
              id: "nemo.unknownError",
              // Default Message in English
            })
          );
        });
    } else if (actionType === appConstants.OFFER_STATUS_ACTION.UNPUBLISH) {
      updateOffer(offerInfo.offerId, appConstants.OFFER_STATUS.UNPUBLISHED)
        .then((response) => {
          setLoading(false);
          if (window.getValue(response, "success")) {
            message.success(
              intl.formatMessage({
                // Matching ID as above
                id: "nemo.unpublishedOfferSuccessfully!",
                // Default Message in English
              })
            );
            hideModal(true);
            setVisible(false);
          } else {
            if (window.getValue(response, "error.message")) {
              message.error(window.getValue(response, "error.message"));
            } else {
              message.error(
                intl.formatMessage({
                  // Matching ID as above
                  id: "nemo.failedTounpublishOffer",
                  // Default Message in English
                })
              );
            }
          }
        })
        .catch((error) => {
          console.log("error = ", error.toString());
          message.error({
            // Matching ID as above
            id: "nemo.unknownError",
            // Default Message in English
          });
        });
    } else {
      hideModal(false);
      setVisible(false);
    }
  };

  const cancelModal = () => {
    hideModal();
    setVisible(false);
  };

  return (
    <Modal
      title={
        <React.Fragment>
          <InfoCircleOutlined className={styles["cart-popup-info-icon"]} />
          {actionType === appConstants.OFFER_STATUS_ACTION.PUBLISH ? (
            <FormattedMessage id="nemo.titleChangeToPublish" />
          ) : actionType === appConstants.OFFER_STATUS_ACTION.UNPUBLISH ? (
            <FormattedMessage id="nemo.titleChangeToUnPublish" />
          ) : actionType === appConstants.OFFER_STATUS_ACTION.ARCHIVE ? (
            <FormattedMessage id="nemo.titleChangeToArchive" />
          ) : (
            <Fragment></Fragment>
          )}
        </React.Fragment>
      }
      onCancel={cancelModal}
      visible={visible}
      data-testid="confirm-remove-supplier-modal"
      okText={<FormattedMessage id="nemo.yes" />}
      cancelText={<FormattedMessage id="nemo.cancel" />}
      footer={[
        <Button
          key="back"
          onClick={cancelModal}
          data-testid="change-offer-cancel-btn"
        >
          <FormattedMessage id="nemo.cancel" />
        </Button>,
        <Button
          key="submit"
          type="primary"
          loading={loading}
          danger={
            actionType === appConstants.OFFER_STATUS_ACTION.UNPUBLISH ||
            actionType === appConstants.OFFER_STATUS_ACTION.ARCHIVE ||
            offerInfo.status.name === appConstants.OFFER_STATUS.ARCHIVED
          }
          onClick={handleOk}
          data-testid="change-offer-yes-btn"
        >
          <FormattedMessage id="nemo.yes" />
        </Button>,
      ]}
    >
      <Fragment>
        {(offerInfo.status.name === appConstants.OFFER_STATUS.UNPUBLISHED ||
          offerInfo.status.name === appConstants.OFFER_STATUS.ARCHIVED) &&
        actionType === appConstants.OFFER_STATUS_ACTION.PUBLISH ? (
          <p data-testid="publish-offer-change-str">
            <FormattedMessage id="nemo.publishOfferStatusChangeImpact" />
          </p>
        ) : (
          <Fragment />
        )}
        {offerInfo.status.name === appConstants.OFFER_STATUS.PUBLISHED &&
        actionType === appConstants.OFFER_STATUS_ACTION.UNPUBLISH ? (
          <p data-testid="unpublish-offer-change-str">
            <FormattedMessage id="nemo.unpublishOfferStatusChangeImpact" />
          </p>
        ) : (
          <Fragment />
        )}
        {offerInfo.status.name === appConstants.OFFER_STATUS.ARCHIVED &&
        actionType === appConstants.OFFER_STATUS_ACTION.UNPUBLISH ? (
          <p data-testid="archive-unpublish-offer-change-str">
            <FormattedMessage id="nemo.unpublishOfferFromArchiveStatusChangeImpact" />
          </p>
        ) : (
          <Fragment />
        )}
        {offerInfo.status.name === appConstants.OFFER_STATUS.UNPUBLISHED &&
        actionType === appConstants.OFFER_STATUS_ACTION.ARCHIVE ? (
          <p data-testid="unpublish-archive-offer-change-str">
            <FormattedMessage id="nemo.unpublishOfferArchiveImpact" />
          </p>
        ) : (
          <Fragment />
        )}
      </Fragment>
    </Modal>
  );
}
