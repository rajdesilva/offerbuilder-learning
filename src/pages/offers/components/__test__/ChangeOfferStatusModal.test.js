import React from "react";
import { createStore } from "redux";
import { render, cleanup } from "../../../../helpers/testUtils";
import { MemoryRouter } from "react-router-dom";
import ChangeOfferStatusModal from "../ChangeOfferStatusModal";

import { tempStore } from "./tempStore";

import { appConstants } from "../../../../common";

window.fetchWrapper = window.fetchWrapper = jest.fn().mockResolvedValue({
  success: true,
});

afterEach(cleanup);

describe("ChangeOfferStatusModal component test", () => {
  test("create ChangeOfferStatusModal components snapshot", () => {
    const store = createStore(() => ({ ...tempStore }));
    store.dispatch = jest.fn();
    const { container } = render(
      <MemoryRouter>
        <ChangeOfferStatusModal
          offerInfo={{
            bookingDateRange: {
              startDate: "2021-02-02",
              endDate: "2021-02-06",
            },
            bookingDatesTimezone: { value: "Europe/Amsterdam" },
            brandName: "Client Demo Brand",
            createdDate: "02.02.2021",
            currency: "EUR",
            deepLinkSettingsInfo: {
              destination: {
                lat: 48.2081743,
                lng: 16.3738189,
                city: "Pune",
              },
              adultOccupancy: 1,
              childOccupancy: 0,
              checkInType: "rolling",
              includeAllProperties: true,
            },
            displayBookingDateRange: {
              startDate: "02.02.2021",
              endDate: "06.02.2021",
            },
            displayTravellingDateRange: {
              startDate: "02.02.2021",
              endDate: "06.02.2021",
            },
            images: [],
            key: "02Feb2021_1833",
            languages: ["EN"],
            lcn: false,
            name: "PKM Sacher Offer",
            offerId: "02Feb2021_1833",
            status: { id: "PUBLISHED", name: "PUBLISHED" },
            storefrontName: "PC Travel",
            travellingDateRange: {
              startDate: "2021-02-02",
              endDate: "2021-02-06",
            },
            id: "16118406",
            internalName: "Cypress Offer - Edit",
            currencyCode: "EUR",
            marketingInfo: [
              {
                title: "JWM Start Hotels",
                lang: "EN",
                shortDescription: "Experience award-winning service",
                description:
                  "Experience award-winning service and sophisticated style at JW Marriott Hotel Vienna.",
                termsAndConditions: "Standard terms and conditions",
              },
            ],
            properties: [
              {
                internalId: 1199,
                propertyCode: "DEMO_BANJARE",
                name: "BANJARE & BANJARE STAR HOTEL",
                channel: "DEMO_OFFERBUILDER",
                supplier: "ntp",
                city: "Vienna",
                rating: 4,
                ratingProvider: "",
                remainingCapitalPool: 0,
                lcn: false,
                images: [],
                marketingInfo: [
                  {
                    lang: "EN",
                    shortDescription: "Experience award-winning service",
                    description:
                      "Experience award-winning service and sophisticated style at JW Marriott Hotel Vienna.",
                  },
                ],
              },
            ],
            brands: [
              {
                id: "idb",
                name: "Internal Demo Brand",
                storefronts: [
                  {
                    id: "ids2",
                    name: "Best Travel",
                    suppliers: [
                      {
                        id: "ntp",
                        name: "NTP",
                        channels: [
                          {
                            id: "DEMO_OFFERBUILDER",
                            name: "DEMO_OFFERBUILDER",
                          },
                        ],
                      },
                    ],
                  },
                ],
              },
            ],
            createdAt: "2021-01-28T13:30:33.188+0000",
            updatedAt: "2021-01-28T14:01:21.818+0000",
            bookingStartDate: null,
            bookingEndDate: null,
            bookingZoneId: null,
            travelStartDate: null,
            travelEndDate: null,
          }}
          actionType={appConstants.OFFER_STATUS_ACTION.PUBLISH}
          hideModal={() => {}}
        />
      </MemoryRouter>,
      {
        store,
      }
    );

    expect(container.firstChild).toMatchSnapshot();
  });
  test("ChangeOfferStatusModal components- Change offer status to PUBLISH ", () => {
    const store = createStore(() => ({ ...tempStore }));
    store.dispatch = jest.fn();
    const hideModal = (value) => jest.fn();
    const { getByText } = render(
      <MemoryRouter>
        <ChangeOfferStatusModal
          offerInfo={{
            bookingDateRange: {
              startDate: "2021-02-02",
              endDate: "2021-02-06",
            },
            bookingDatesTimezone: { value: "Europe/Amsterdam" },
            brandName: "Client Demo Brand",
            createdDate: "02.02.2021",
            currency: "EUR",
            deepLinkSettingsInfo: {
              destination: {
                lat: 48.2081743,
                lng: 16.3738189,
                city: "Pune",
              },
              adultOccupancy: 1,
              childOccupancy: 0,
              checkInType: "rolling",
              includeAllProperties: true,
            },
            displayBookingDateRange: {
              startDate: "02.02.2021",
              endDate: "06.02.2021",
            },
            displayTravellingDateRange: {
              startDate: "02.02.2021",
              endDate: "06.02.2021",
            },
            images: [],
            key: "02Feb2021_1833",
            languages: ["EN"],
            lcn: false,
            name: "PKM Sacher Offer",
            offerId: "02Feb2021_1833",
            status: { id: "UNPUBLISHED", name: "UNPUBLISHED" },
            storefrontName: "PC Travel",
            travellingDateRange: {
              startDate: "2021-02-02",
              endDate: "2021-02-06",
            },
            id: "16118406",
            internalName: "Cypress Offer - Edit",
            currencyCode: "EUR",
            marketingInfo: [
              {
                title: "JWM Start Hotels",
                lang: "EN",
                shortDescription: "Experience award-winning service",
                description:
                  "Experience award-winning service and sophisticated style at JW Marriott Hotel Vienna.",
                termsAndConditions: "Standard terms and conditions",
              },
            ],
            properties: [
              {
                internalId: 1199,
                propertyCode: "DEMO_BANJARE",
                name: "BANJARE & BANJARE STAR HOTEL",
                channel: "DEMO_OFFERBUILDER",
                supplier: "ntp",
                city: "Vienna",
                rating: 4,
                ratingProvider: "",
                remainingCapitalPool: 0,
                lcn: false,
                images: [],
                marketingInfo: [
                  {
                    lang: "EN",
                    shortDescription: "Experience award-winning service",
                    description:
                      "Experience award-winning service and sophisticated style at JW Marriott Hotel Vienna.",
                  },
                ],
              },
            ],
            brands: [
              {
                id: "idb",
                name: "Internal Demo Brand",
                storefronts: [
                  {
                    id: "ids2",
                    name: "Best Travel",
                    suppliers: [
                      {
                        id: "ntp",
                        name: "NTP",
                        channels: [
                          {
                            id: "DEMO_OFFERBUILDER",
                            name: "DEMO_OFFERBUILDER",
                          },
                        ],
                      },
                    ],
                  },
                ],
              },
            ],
            createdAt: "2021-01-28T13:30:33.188+0000",
            updatedAt: "2021-01-28T14:01:21.818+0000",
            bookingStartDate: null,
            bookingEndDate: null,
            bookingZoneId: null,
            travelStartDate: null,
            travelEndDate: null,
          }}
          actionType={appConstants.OFFER_STATUS_ACTION.PUBLISH}
          hideModal={hideModal}
        />
      </MemoryRouter>,
      {
        store,
      }
    );
    expect(getByText("Do You Want To Publish This Offer?")).toBeInTheDocument();
    expect(
      getByText(
        "Depending on where offer is to show, it may not show. To show, action may be needed where to show. Do you wish to proceed?"
      )
    ).toBeInTheDocument();
  });
  test("ChangeOfferStatusModal components- Change offer status to UNPUBLISH ", () => {
    const store = createStore(() => ({ ...tempStore }));
    store.dispatch = jest.fn();
    const hideModal = (value) => jest.fn();
    const { getByText, queryByTestId } = render(
      <MemoryRouter>
        <ChangeOfferStatusModal
          offerInfo={{
            bookingDateRange: {
              startDate: "2021-02-02",
              endDate: "2021-02-06",
            },
            bookingDatesTimezone: { value: "Europe/Amsterdam" },
            brandName: "Client Demo Brand",
            createdDate: "02.02.2021",
            currency: "EUR",
            deepLinkSettingsInfo: {
              destination: {
                lat: 48.2081743,
                lng: 16.3738189,
                city: "Pune",
              },
              adultOccupancy: 1,
              childOccupancy: 0,
              checkInType: "rolling",
              includeAllProperties: true,
            },
            displayBookingDateRange: {
              startDate: "02.02.2021",
              endDate: "06.02.2021",
            },
            displayTravellingDateRange: {
              startDate: "02.02.2021",
              endDate: "06.02.2021",
            },
            images: [],
            key: "02Feb2021_1833",
            languages: ["EN"],
            lcn: false,
            name: "PKM Sacher Offer",
            offerId: "02Feb2021_1833",
            status: { id: "PUBLISHED", name: "PUBLISHED" },
            storefrontName: "PC Travel",
            travellingDateRange: {
              startDate: "2021-02-02",
              endDate: "2021-02-06",
            },
            id: "16118406",
            internalName: "Cypress Offer - Edit",
            currencyCode: "EUR",
            marketingInfo: [
              {
                title: "JWM Start Hotels",
                lang: "EN",
                shortDescription: "Experience award-winning service",
                description:
                  "Experience award-winning service and sophisticated style at JW Marriott Hotel Vienna.",
                termsAndConditions: "Standard terms and conditions",
              },
            ],
            properties: [
              {
                internalId: 1199,
                propertyCode: "DEMO_BANJARE",
                name: "BANJARE & BANJARE STAR HOTEL",
                channel: "DEMO_OFFERBUILDER",
                supplier: "ntp",
                city: "Vienna",
                rating: 4,
                ratingProvider: "",
                remainingCapitalPool: 0,
                lcn: false,
                images: [],
                marketingInfo: [
                  {
                    lang: "EN",
                    shortDescription: "Experience award-winning service",
                    description:
                      "Experience award-winning service and sophisticated style at JW Marriott Hotel Vienna.",
                  },
                ],
              },
            ],
            brands: [
              {
                id: "idb",
                name: "Internal Demo Brand",
                storefronts: [
                  {
                    id: "ids2",
                    name: "Best Travel",
                    suppliers: [
                      {
                        id: "ntp",
                        name: "NTP",
                        channels: [
                          {
                            id: "DEMO_OFFERBUILDER",
                            name: "DEMO_OFFERBUILDER",
                          },
                        ],
                      },
                    ],
                  },
                ],
              },
            ],
            createdAt: "2021-01-28T13:30:33.188+0000",
            updatedAt: "2021-01-28T14:01:21.818+0000",
            bookingStartDate: null,
            bookingEndDate: null,
            bookingZoneId: null,
            travelStartDate: null,
            travelEndDate: null,
          }}
          actionType={appConstants.OFFER_STATUS_ACTION.UNPUBLISH}
          hideModal={hideModal}
        />
      </MemoryRouter>,
      {
        store,
      }
    );

    expect(
      getByText("Do You Want To UnPublish This Offer?")
    ).toBeInTheDocument();
    expect(
      queryByTestId("unpublish-archive-offer-change-str")
    ).not.toBeInTheDocument();
    expect(queryByTestId("unpublish-offer-change-str")).toBeInTheDocument();
    expect(queryByTestId("publish-offer-change-str")).not.toBeInTheDocument();
    expect(
      queryByTestId("archive-unpublish-offer-change-str")
    ).not.toBeInTheDocument();
  });
  test("ChangeOfferStatusModal components- Change offer status to ARCHIVE ", () => {
    const store = createStore(() => ({ ...tempStore }));
    store.dispatch = jest.fn();
    const hideModal = (value) => jest.fn();
    const { getByText, queryByTestId } = render(
      <MemoryRouter>
        <ChangeOfferStatusModal
          offerInfo={{
            bookingDateRange: {
              startDate: "2021-02-02",
              endDate: "2021-02-06",
            },
            bookingDatesTimezone: { value: "Europe/Amsterdam" },
            brandName: "Client Demo Brand",
            createdDate: "02.02.2021",
            currency: "EUR",
            deepLinkSettingsInfo: {
              destination: {
                lat: 48.2081743,
                lng: 16.3738189,
                city: "Pune",
              },
              adultOccupancy: 1,
              childOccupancy: 0,
              checkInType: "rolling",
              includeAllProperties: true,
            },
            displayBookingDateRange: {
              startDate: "02.02.2021",
              endDate: "06.02.2021",
            },
            displayTravellingDateRange: {
              startDate: "02.02.2021",
              endDate: "06.02.2021",
            },
            images: [],
            key: "02Feb2021_1833",
            languages: ["EN"],
            lcn: false,
            name: "PKM Sacher Offer",
            offerId: "02Feb2021_1833",
            status: { id: "PUBLISHED", name: "PUBLISHED" },
            storefrontName: "PC Travel",
            travellingDateRange: {
              startDate: "2021-02-02",
              endDate: "2021-02-06",
            },
            id: "16118406",
            internalName: "Cypress Offer - Edit",
            currencyCode: "EUR",
            marketingInfo: [
              {
                title: "JWM Start Hotels",
                lang: "EN",
                shortDescription: "Experience award-winning service",
                description:
                  "Experience award-winning service and sophisticated style at JW Marriott Hotel Vienna.",
                termsAndConditions: "Standard terms and conditions",
              },
            ],
            properties: [
              {
                internalId: 1199,
                propertyCode: "DEMO_BANJARE",
                name: "BANJARE & BANJARE STAR HOTEL",
                channel: "DEMO_OFFERBUILDER",
                supplier: "ntp",
                city: "Vienna",
                rating: 4,
                ratingProvider: "",
                remainingCapitalPool: 0,
                lcn: false,
                images: [],
                marketingInfo: [
                  {
                    lang: "EN",
                    shortDescription: "Experience award-winning service",
                    description:
                      "Experience award-winning service and sophisticated style at JW Marriott Hotel Vienna.",
                  },
                ],
              },
            ],
            brands: [
              {
                id: "idb",
                name: "Internal Demo Brand",
                storefronts: [
                  {
                    id: "ids2",
                    name: "Best Travel",
                    suppliers: [
                      {
                        id: "ntp",
                        name: "NTP",
                        channels: [
                          {
                            id: "DEMO_OFFERBUILDER",
                            name: "DEMO_OFFERBUILDER",
                          },
                        ],
                      },
                    ],
                  },
                ],
              },
            ],
            createdAt: "2021-01-28T13:30:33.188+0000",
            updatedAt: "2021-01-28T14:01:21.818+0000",
            bookingStartDate: null,
            bookingEndDate: null,
            bookingZoneId: null,
            travelStartDate: null,
            travelEndDate: null,
          }}
          actionType={appConstants.OFFER_STATUS_ACTION.ARCHIVE}
          hideModal={hideModal}
        />
      </MemoryRouter>,
      {
        store,
      }
    );

    expect(getByText("Do You Want To Archive This Offer?")).toBeInTheDocument();
    expect(
      queryByTestId("unpublish-archive-offer-change-str")
    ).not.toBeInTheDocument();
    expect(queryByTestId("unpublish-offer-change-str")).not.toBeInTheDocument();
    expect(queryByTestId("publish-offer-change-str")).not.toBeInTheDocument();
    expect(
      queryByTestId("archive-unpublish-offer-change-str")
    ).not.toBeInTheDocument();
  });
  test("ChangeOfferStatusModal components- Change offer status to UNPUBLISH WHEN ARCHIVE OFFER", () => {
    const store = createStore(() => ({ ...tempStore }));
    store.dispatch = jest.fn();
    const hideModal = (value) => jest.fn();
    const { getByText, queryByTestId } = render(
      <MemoryRouter>
        <ChangeOfferStatusModal
          offerInfo={{
            bookingDateRange: {
              startDate: "2021-02-02",
              endDate: "2021-02-06",
            },
            bookingDatesTimezone: { value: "Europe/Amsterdam" },
            brandName: "Client Demo Brand",
            createdDate: "02.02.2021",
            currency: "EUR",
            deepLinkSettingsInfo: {
              destination: {
                lat: 48.2081743,
                lng: 16.3738189,
                city: "Pune",
              },
              adultOccupancy: 1,
              childOccupancy: 0,
              checkInType: "rolling",
              includeAllProperties: true,
            },
            displayBookingDateRange: {
              startDate: "02.02.2021",
              endDate: "06.02.2021",
            },
            displayTravellingDateRange: {
              startDate: "02.02.2021",
              endDate: "06.02.2021",
            },
            images: [],
            key: "02Feb2021_1833",
            languages: ["EN"],
            lcn: false,
            name: "PKM Sacher Offer",
            offerId: "02Feb2021_1833",
            status: { id: "UNPUBLISHED", name: "UNPUBLISHED" },
            storefrontName: "PC Travel",
            travellingDateRange: {
              startDate: "2021-02-02",
              endDate: "2021-02-06",
            },
            id: "16118406",
            internalName: "Cypress Offer - Edit",
            currencyCode: "EUR",
            marketingInfo: [
              {
                title: "JWM Start Hotels",
                lang: "EN",
                shortDescription: "Experience award-winning service",
                description:
                  "Experience award-winning service and sophisticated style at JW Marriott Hotel Vienna.",
                termsAndConditions: "Standard terms and conditions",
              },
            ],
            properties: [
              {
                internalId: 1199,
                propertyCode: "DEMO_BANJARE",
                name: "BANJARE & BANJARE STAR HOTEL",
                channel: "DEMO_OFFERBUILDER",
                supplier: "ntp",
                city: "Vienna",
                rating: 4,
                ratingProvider: "",
                remainingCapitalPool: 0,
                lcn: false,
                images: [],
                marketingInfo: [
                  {
                    lang: "EN",
                    shortDescription: "Experience award-winning service",
                    description:
                      "Experience award-winning service and sophisticated style at JW Marriott Hotel Vienna.",
                  },
                ],
              },
            ],
            brands: [
              {
                id: "idb",
                name: "Internal Demo Brand",
                storefronts: [
                  {
                    id: "ids2",
                    name: "Best Travel",
                    suppliers: [
                      {
                        id: "ntp",
                        name: "NTP",
                        channels: [
                          {
                            id: "DEMO_OFFERBUILDER",
                            name: "DEMO_OFFERBUILDER",
                          },
                        ],
                      },
                    ],
                  },
                ],
              },
            ],
            createdAt: "2021-01-28T13:30:33.188+0000",
            updatedAt: "2021-01-28T14:01:21.818+0000",
            bookingStartDate: null,
            bookingEndDate: null,
            bookingZoneId: null,
            travelStartDate: null,
            travelEndDate: null,
          }}
          actionType={appConstants.OFFER_STATUS_ACTION.ARCHIVE}
          hideModal={hideModal}
        />
      </MemoryRouter>,
      {
        store,
      }
    );

    expect(getByText("Do You Want To Archive This Offer?")).toBeInTheDocument();
    expect(
      queryByTestId("unpublish-archive-offer-change-str")
    ).toBeInTheDocument();
    expect(queryByTestId("unpublish-offer-change-str")).not.toBeInTheDocument();
    expect(queryByTestId("publish-offer-change-str")).not.toBeInTheDocument();
    expect(
      queryByTestId("archive-unpublish-offer-change-str")
    ).not.toBeInTheDocument();
  });
});
