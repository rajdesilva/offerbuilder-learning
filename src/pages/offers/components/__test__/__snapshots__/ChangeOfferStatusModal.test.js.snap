// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeOfferStatusModal component test create ChangeOfferStatusModal components snapshot 1`] = `null`;
